drivers/cleanup.py

```diff
@@ -585,11 +589,19 @@ def isLeafCoalesceable(self):
                 not self.hidden and \
                 len(self.children) == 0
 
-    def canLiveCoalesce(self):
+    def canLiveCoalesce(self, speed):
         """Can we stop-and-leaf-coalesce this VDI? The VDI must be
         isLeafCoalesceable() already"""
-        return self.getSizeVHD() <= self.LIVE_LEAF_COALESCE_MAX_SIZE or \
-                self.getConfig(self.DB_LEAFCLSC) == self.LEAFCLSC_FORCE
+        feasibleSize = False
+        if speed:
+            feasibleSize =\
+                self.getSizeVHD()/speed < self.LIVE_LEAF_COALESCE_TIMEOUT
```

Comment:
I think we should be conservative here and divide the timeout by 2 to make sure that we get done within the time cutoff

Reply:
fixed